src/Microsoft.Health.Fhir.CosmosDb.UnitTests/Features/Storage/CosmosResponseProcessorTests.cs

```diff
@@ -156,19 +156,109 @@ public async Task GivenADocumentClientExceptionWithSpecificMessage_WhenProcessin
         }
 
         [Fact]
-        public async Task GivenADocumentClientExceptionWithCustomerManagedKeyInaccessibleSubStatus_WhenProcessing_ThenExceptionShouldThrow()
+        public async Task GivenADocumentClientExceptionWithCmkAadClientCredentialsGrantFailureSubStatus_WhenProcessing_ThenExceptionShouldThrow()
         {
-            DocumentClientException documentClientException = CreateDocumentClientException("12.4", "fail", HttpStatusCode.Forbidden, CosmosDbSubStatusValues.CustomerManagedKeyInaccessible.ToString());
+            DocumentClientException documentClientException = CreateDocumentClientException("12.4", "fail", HttpStatusCode.Forbidden, CosmosDbSubStatusValues.CmkAadClientCredentialsGrantFailure.ToString());
 
-            await Assert.ThrowsAsync<CustomerManagedKeyInaccessibleException>(async () => await _cosmosResponseProcessor.ProcessException(documentClientException));
+            await Assert.ThrowsAsync<CmkAadClientCredentialsGrantFailureException>(async () => await _cosmosResponseProcessor.ProcessException(documentClientException));
+
+            ValidateExecution(expectedSessionToken: null, 12.4, false);
+        }
+
+        [Fact]
+        public async Task GivenADocumentClientExceptionWithCmkAadServiceUnavailableSubStatus_WhenProcessing_ThenExceptionShouldThrow()
```

Comment:
nit: Since all these tests differ only in the sub-status code that is passed to `CreateDocumentClientException`, you can just make the test a `[Theory]` and pass the different codes as `InlineData`. Similar to the one below: `GivenADocumentClientExceptionWithForbiddenStatusCodeAndUnknownSubStatus_WhenProcessing_ThenNothingElseShouldOccur`

Reply:
Never mind, just realized that it checks for a different exception in every test.

Reply:
Per your other suggestion, I now have only one exception.  Thus, I was able to use the [Theory].  Thanks!